HousingManagementSystemApi/Repositories/UniversalHousingAddressesRepository.cs

```diff
@@ -0,0 +1,59 @@
+namespace HousingManagementSystemApi.Repositories
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Data;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Ardalis.GuardClauses;
+    using Dapper;
+    using HACT.Dtos;
+    using JetBrains.Annotations;
+    using Models;
+
+    public class UniversalHousingAddressesRepository : IAddressesRepository
+    {
+        private Func<IDbConnection> dbConnectionFunc;
+
+        public UniversalHousingAddressesRepository([NotNull] Func<IDbConnection> dbConnectionFunc)
+        {
+            Guard.Against.Null(dbConnectionFunc, nameof(dbConnectionFunc));
+
+            this.dbConnectionFunc = dbConnectionFunc;
+        }
+
+        public async Task<IEnumerable<PropertyAddress>> GetAddressesByPostcode([NotNull] string postcode)
+        {
+            Guard.Against.NullOrWhiteSpace(postcode, nameof(postcode));
+
+            var sql = @$"SELECT pr.prop_ref AS PropertyReference
+                ,pr.u_nlpguprn AS Uprn
+                ,RTRIM(pr.post_desig) AS PostDesig
+                ,RTRIM(REPLACE(pr.short_address,rtrim(pr.post_desig),'')) AS ShortAddress
+                ,RTRIM(pr.post_code) as PostCode
+                FROM property AS pr
+                --rent details
+                    LEFT OUTER JOIN rent AS r
+                    ON pr.prop_ref = r.prop_ref
+                --tenancy
+                    LEFT OUTER JOIN tenagree AS ten
+                    ON pr.prop_ref = ten.prop_ref
+                --property type description
+                    LEFT OUTER JOIN (SELECT [lu_ref],[lu_desc] FROM [lookup] WHERE lu_type ='GPT') AS ptype
+                    ON r.prop_type = ptype.lu_ref
+                WHERE level_code in ('2')
+                AND eot = '1900-01-01 00:00:00'
+                AND (UPPER(REPLACE(post_code, ' ','')) like UPPER(REPLACE('%{postcode}%', ' ','')))
```

Comment:
This is the additional change to [PR 4](https://github.com/City-of-Lincoln-Council/HousingManagementSystemApi/pull/4)

Reply:
If i'm reading this correctly, it uppercasing the postcode in the postcode column. If so are we assuming that the postcode will always be uppercase. If so, then are we only doing that in the frontend, what if another application consumes this API that does not uppercase the postcode. Should we also do it here as a precaution?

Reply:
Yes, this is uppercasing the `post_code` column.
It's also uppercasing the `postcode` input parameter.

So it covers the case where the passed parameter differs from the database value by case.